### PR TITLE
Add Variable functionality for doc

### PR DESCRIPTION
**Pull request by BrunoMine**
I should include `npc.data.*` yet

### CODE REVIEW

**Comment by BrunoMine** [comment]
Okay, this will help a lot of new developers.

**Comment by BrunoMine** [comment]
Ready to merge everything and launch a new version, soon I will be updating my mod.

File: doc/api.md [hunk 1]
```diff
@@ -212,6 +230,10 @@ For permanent storage, use `npc.data.*` functions.
 * `npc.exec.var.get(luaentity, key_name)`: Returns the value of a given key_name
 * `npc.exec.var.set(luaentity, key_name, new_value)`: Update a value in the execution context
 * `npc.exec.var.remove(luaentity, key_name)`: Remove a variable from the execution context
+* `npc.data.put(luaentity, key_name, value, readonly)`: This function adds a value to the execution context of the current process
```
Comment (hkzorman) [inline doc/api.md]: This says "execution context of the current process". It should say, "permanent data storage in the Lua entity" or similar

File: doc/api.md [hunk 2]
```diff
@@ -212,6 +230,10 @@ For permanent storage, use `npc.data.*` functions.
...
+* `npc.data.get(luaentity, key_name)`: Returns the value of a given key
+* `npc.data.set(luaentity, key_name, new_value)`: This function updates a value in the execution context
```
Comment (hkzorman) [inline doc/api.md]: Same as above

File: doc/api.md [hunk 3]
```diff
@@ -212,6 +230,10 @@ For permanent storage, use `npc.data.*` functions.
...
+* `npc.data.remove(luaentity, key_name)`: This function removes a variable from the execution context
```
Comment (hkzorman) [inline doc/api.md]: Same as above

File: doc/api.md [hunk 4]
```diff
@@ -182,6 +182,24 @@ are some general tips to keep in mind while writing programs
...
+IMPORTANT: These variables are *NOT* deleted. Be careful what you store on it or 
+the NPC object can grow in size very quickly. 
+For temporary storage, use `npc.exec.var.*` functions.
+
+#### Methods
+* `npc.data.put(luaentity, key_name, value, readonly)`: This function adds a value to the execution context of the current process
+  * Readonly defaults to false. 
+  * Returns false if failed due to key_name conflict, or returns true if successful.
+* `npc.data.get(luaentity, key_name)`: Returns the value of a given key. If not found returns nil
+* `npc.data.set(luaentity, key_name, new_value)`: This function updates a value in the execution context
```
Comment (hkzorman) [inline doc/api.md]: Mention of "execution context". Change to "permanent storage" or something.

File: doc/api.md [hunk 5]
```diff
@@ -182,6 +182,24 @@ are some general tips to keep in mind while writing programs
...
+  * Returns false if the value is read-only or if key isn't found.
+  * Returns true if able to update value.
+* `npc.data.remove(luaentity, key_name)`: This function removes a variable from the execution context
```
Comment (hkzorman) [inline doc/api.md]: Same as above.

File: doc/api.md [hunk 6]
```diff
@@ -182,13 +182,58 @@ are some general tips to keep in mind while writing programs
...
 * `npc.programs.execute(luaentity, program_name, {program arguments})`: Execute a program for a NPC
 * `npc.programs.instr.register(name, func)`: Register a instruction
 * `npc.programs.instr.execute(self, name, args)`: Execute a instruction for a NPC
 * `npc.exec.enqueue_program(luaentity, program_name, {program arguments}, interrupt_options, is_state_program)`: Add program to schedule queue
+* `npc.exec.proc.enqueue(luaentity, instruction_name, {instruction arguments})`: Add instruction to process queue
+* `npc.exec.var.put(luaentity, key_name, value, readonly)`: Put a value to the execution context of the current process
+* `npc.exec.var.get(luaentity, key_name)`: Returns the value of a given key_name
+* `npc.exec.var.set(luaentity, key_name, new_value)`: Update a value in the execution context
+* `npc.exec.var.remove(luaentity, key_name)`: Remove a variable from the execution context
+* `npc.data.put(luaentity, key_name, value, readonly)`: This function adds a value to the execution context of the current process
```
Comment (hkzorman) [inline doc/api.md]: Same as above

File: doc/api.md [hunk 7]
```diff
@@ -182,13 +182,58 @@ are some general tips to keep in mind while writing programs
...
+* `npc.data.get(luaentity, key_name)`: Returns the value of a given key
+* `npc.data.set(luaentity, key_name, new_value)`: This function updates a value in the execution context
+* `npc.data.remove(luaentity, key_name)`: This function removes a variable from the execution context
```
Comment (hkzorman) [inline doc/api.md]: Same as above

File: doc/api.md [hunk 8]
```diff
@@ -182,13 +182,58 @@ are some general tips to keep in mind while writing programs
...
 * `npc.programs.instr.execute(self, name, args)`: Execute a instruction for a NPC
 * `npc.exec.enqueue_program(luaentity, program_name, {program arguments}, interrupt_options, is_state_program)`: Add program to schedule queue
+* `npc.exec.proc.enqueue(luaentity, instruction_name, {instruction arguments})`: Add instruction to process queue
+* `npc.exec.var.put(luaentity, key_name, value, readonly)`: Put a value to the execution context of the current process
+* `npc.exec.var.get(luaentity, key_name)`: Returns the value of a given key_name
+* `npc.exec.var.set(luaentity, key_name, new_value)`: Update a value in the execution context
+* `npc.exec.var.remove(luaentity, key_name)`: Remove a variable from the execution context
+* `npc.data.put(luaentity, key_name, value, readonly)`: This function adds a value to the execution context of the current process
+* `npc.data.get(luaentity, key_name)`: Returns the value of a given key
+* `npc.data.set(luaentity, key_name, new_value)`: This function updates a value in the execution context
```
Comment (hkzorman) [inline doc/api.md]: Same as above

**Comment by BrunoMine** [comment]
I was going to talk about it, it looks like it's that way in coding. But I will adjust here in the documentation now.

**Comment by hkzorman** [comment]
Most probably I have to change in code as well, thanks 👍

**Comment by BrunoMine** [comment]
Check if you like it

**Comment by hkzorman** [comment]
👍 Thank you for this!

**Pull request merged**